src/hooks/useStepsContext.test.ts

```diff
@@ -0,0 +1,181 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+import {
+  act,
+  renderHook,
+  RenderHookResult,
+} from "@testing-library/react-hooks";
+import type { ActionContext, Step, Steps } from "../common/types";
+import { IStepsContext } from "../contexts/StepsContext";
+import { useStepsContext } from "./useStepsContext";
+
+describe("useStepsContext", () => {
+  let defaultResult: RenderHookResult<unknown, IStepsContext>;
+  let defaultSteps: Steps;
+
+  beforeEach(async () => {
+    defaultSteps = [
+      [
+        {
+          action: {
+            name: "first-step-1",
+            signals: [],
+          },
+          frameUrl: "https://www.elastic.co",
+          isMainFrame: true,
+          pageAlias: "pageAlias",
+        },
+      ],
+      [
+        {
+          action: {
+            name: "first-step-2",
+            signals: [],
+          },
+          frameUrl: "https://www.elastic.co",
+          isMainFrame: true,
+          pageAlias: "pageAlias",
+        },
+        {
+          action: {
+            name: "second-step-2",
+            signals: [],
+          },
+          frameUrl: "https://www.elastic.co",
+          isMainFrame: true,
+          pageAlias: "pageAlias",
+        },
+      ],
+    ];
```

Comment:
Just a nit: can we wrap the creation of steps into a function so we don't have to remember the object structure every time? There seems to be a lot of repetition here so I imagine that the following would be nice:

```
const createStep = (overrides, index) => {
  return {
    action: {
      name: `step-${index}`,
      signals: [],
    },
    frameUrl: "https://www.elastic.co",
    isMainFrame: true,
    pageAlias: "pageAlias",
    ...verrides
  };
};

const createSteps = (overridesArr) => overridesArr.map(createStep);
```

Reply:
Sure, I agree, I think it's awful as-is 😆

Reply:
There's a follow-up PR behind this one that makes it much worse, adding a create function will be much nicer.

Reply:
Fixed, check feda1bb4b34965db7c6dad46e6f5643e4e50dbbd.